### PR TITLE
Feature/burmark1/haloexchange workgroup

### PR DESCRIPTION
**Pull request by MrBurmark**
Add cuda and hip workgroup variants to halo exchange loop. Note that the hip workgroup variant is conditionally compiled in based on the RAJA configuration.

### CODE REVIEW

File: src/apps/HALOEXCHANGE-Cuda.cpp [hunk 1]
```diff
@@ -163,6 +164,86 @@ void HALOEXCHANGE::runCudaVariant(VariantID vid)
 
     HALOEXCHANGE_DATA_TEARDOWN_CUDA;
 
+  } else if ( vid == RAJA_WORKGROUP_CUDA ) {
```
Comment (rhornung67) [inline src/apps/HALOEXCHANGE-Cuda.cpp:167]: Is there a reason to add a new variant like this? Can't it just be RAJA_CUDA?

Comment (MrBurmark) [reply src/apps/HALOEXCHANGE-Cuda.cpp:167]: You could do that. The base cuda version is there to compare against.
I was trying to have a kind of before and after with both Raja variants.

Comment (MrBurmark) [reply src/apps/HALOEXCHANGE-Cuda.cpp:167]: All its really showing between the BASE_CUDA and RAJA_CUDA variants currently is any overhead added by RAJA.

Comment (rhornung67) [reply src/apps/HALOEXCHANGE-Cuda.cpp:167]: what are the two RAJA variants?

Comment (MrBurmark) [reply src/apps/HALOEXCHANGE-Cuda.cpp:167]: The RAJA_CUDA version launches one kernel per variable per message in the same manner as the BASE_CUDA variant.
The RAJA_WORKGROUP_CUDA variant uses kernel fusion to launch only one kernel.

Comment (rhornung67) [reply src/apps/HALOEXCHANGE-Cuda.cpp:167]: OK. Got it. Thanks.

**Review by rhornung67 — DISMISSED** [review]

**Review by rhornung67 — APPROVED** [review]

**Pull request merged**